frontend/src/scenes/surveys/Survey.tsx

```diff
@@ -248,6 +248,39 @@ export function SurveyForm({ id }: { id: string }): JSX.Element {
                             </Group>
                         )
                     )}
+                    <LemonDivider />
+                    <Field name="appearance" label="Thank you message (optional)">
```

Comment:
it seems highly unusual to use kea-forms like this, where we assume it's the entire `appearance` field and then extend values based on the current `appearance`.

I think what you're looking for here is `Group` to get to the right parameter:

```
// using <Group /> to scope the next fields
        <Group key={index} name={['filters', 'multivariate', 'variants', index]}>
          <Field name="name">
            {/* This will update featureFlag.filters.multivariate.variants[index].name */}
            {/* If you don't specify ({ value, onChange }) => (), we add these two props automatically */}
            <Input placeholder="Description" />
          </Field>
        </Group>
```

https://keajs.org/docs/plugins/forms/

because if something else changes value while we're doing this, it can easily bork things

Reply:
but fine to refactor in a follow up, as I assume other appearance fields need the same 👀

Reply:
Well group doesn't exactly do it either because the fields need to be placed before and after other fields. Arguably we can say that maybe the "thanks" options shouldn't belong under the appearances field then if they don't make sense to be grouped together in the form, but they don't belong in the conditions field either IMO, so 🤷‍♀️ 

I think a later surveys UI/UX creation form refactor should fix this though 😄

Reply:
>  because the fields need to be placed before and after other fields.

What do you mean by this?

As far as I understand, group doesn't have any UI effect, and you can use a group multiple times in separate parts of the UI, so where a field appears shouldn't affect this